File: Programmers/level2/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    
    div=list( i for i in range(2,total) if (total%i==0))
    div.reverse()
    div= list([div[i],div[-(i+1)]] for i in range(-(-len(div)//2)) if div[-(i+1)]>2 )
    
    answer = list( d for d in div if (2*d[0] + 2*d[1] == brown+4))
    
    return answer[0]

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        int total=brown+yellow;
        for(int h=3;h*h<=total;h++){
            if(total%h!=0) continue;
            int w=total/h;
            if((w*2)+(h*2-4)==brown) {
                answer=new int[]{w,h};
                break;
            }
        }
        return answer;
    }

}
